packages/app/src/app/store/modules/server/actions.js

```diff
@@ -12,9 +13,67 @@ export function restartContainer({ executor }) {
  * We need to remove this logic later! We need to move it to the terminal component
  * and let that component listen to the executor directly.
  */
-export function logSandboxMessage({ props }) {
-  dispatch({
+export function logSandboxMessage({ props, codeSandboxApi }) {
+  codeSandboxApi.dispatch({
     type: 'terminal:message',
     data: props.data.data,
   });
 }
+
+export function closeSocket({ executor }) {
```

Comment:
@CompuIves This can be an arrow function

```suggestion
export const closeSocket = ({ executor }) => {
```